floris/simulation/flow_field.py

```diff
@@ -66,6 +66,22 @@ class FlowField(BaseClass):
         init=False, factory=lambda: np.array([])
     )
 
+    @turbulence_intensities.validator
+    def turbulence_intensities_validator(
+        self, instance: attrs.Attribute, value: NDArrayFloat
+    ) -> None:
+        try:
+            # Check the turbulence intensity is either length 1 or n_findex
+            if len(value) != 1 and len(value) != self.n_findex:
+                raise ValueError("turbulence_intensities should either be length 1 or n_findex")
+        except TypeError as te:
```

Comment:
What is the TypeError that would be caught here? Note that [attrs converters](https://www.attrs.org/en/stable/init.html#converters) are run before validators, so `value` here is guaranteed to be a `np.array`.

Reply:
I was having trouble catching the issue if you pass something through that doesn't have a len operation, even if it is of type np.array.  The example I found was that I would pass in a float, this would be automatically be upped to an np.array, but it would be a special case without a len operation, calling len(value) produced an error that was not screened out by confirming it is of type np.array

Reply:
Does the change in https://github.com/NREL/floris/pull/782/commits/598aa81293b61ceb5c7c340b68a8a2d533ff4086 affect this?

Reply:
I think it does, if the floris_array_converter comes before this one, then there is no possibility that the len() call could generate a TypeError, would you like me to remove that try/except code on those grounds?

Reply:
Yep that would be great

Reply:
Or more specifically, check if your previous commit affects this code and update as needed

Reply:
no problem, this is actually pretty easy to do, there is now a specific check of this case in test_reinitailize_ti (line 258), where it passes a float in, you can see that the expected TypeError is raised before this code is executed